src/smt/proof_post_processor_dsl.cpp

```diff
@@ -109,7 +60,34 @@ bool ProofPostprocessDsl::shouldUpdate(std::shared_ptr<ProofNode> pn,
                                        bool& continueUpdate)
 {
   ProofRule id = pn->getRule();
-  return id == ProofRule::TRUST || id == ProofRule::TRUST_THEORY_REWRITE;
+  continueUpdate = true;
+  // we should update if we
+  // - Have rule TRUST or TRUST_THEORY_REWRITE,
+  // - We have no premises
+  // - We are not already recursively expanding >= 3 steps of the above form.
+  // We check for the third criteria by tracking a d_traversing vector.
+  if ((id == ProofRule::TRUST || id == ProofRule::TRUST_THEORY_REWRITE)
+      && pn->getChildren().empty() && d_traversing.size() < 3)
+  {
+    Trace("pp-dsl-process") << "...push " << pn.get() << std::endl;
+    d_traversing.push_back(pn);
+    return true;
+  }
+  return false;
+}
+
+bool ProofPostprocessDsl::shouldUpdatePost(std::shared_ptr<ProofNode> pn,
+                                           const std::vector<Node>& fa)
+{
+  // clean up d_traversing at post-traversal
+  // note we may have pushed multiple copies of pn consecutively if a proof
+  // was updated to another trust step.
+  while (!d_traversing.empty() && d_traversing.back() == pn)
+  {
+    Trace("pp-dsl-process") << "...pop " << pn.get() << std::endl;
+    d_traversing.pop_back();
+  }
+  return false;
```

Comment:
Method always returns `false`

Reply:
This is intentional, this is a virtual method that is called where "false" indicates whether we need to update the proof node (again, after its children changed), in this case we do not.